src/Parser/BalanceDruid/Modules/Features/AstralPower.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import Icon from 'common/Icon';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import Module from 'Parser/Core/Module';
+import SPELLS from 'common/SPELLS';
+
+class AstralPower extends Module {
+  
+  lastAstral = 0;
+  aspWasted = 0;
+
+  on_toPlayer_energize(event) {
+      for (let i = 0; i < event.classResources.length; i++) {
+          if (event.classResources[i].type === 8){
+              const maxAsP = event.classResources[i].max / 10;
```

Comment:
Is 10 the max astral power you can have at a time? I think it would be clean to extract this magic number too.

Reply:
Repeated below

Reply:
Nope, in some cases in the objects the Resource (Astral Power in this case) comes multiplied by 10 to avoid decimals (for instance, when the AsP generation is affected by a multiplier that makes it 12.5 instead of 10). So its needed to be divided by 10 on those cases, so everything is in the same base.

Reply:
Ohh ok my bad then. Would it be a big deal to change use ints and transform it only once you display the values? This would prevent floating point errors (although I doubt we'd actually run into issues with that) and be consistent with other spec implementations.

Reply:
I can multiply every non multiplid value by 10 and work with that, no problem :)